src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/BoundedQueueTest.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+@file:Suppress("unused")
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import kotlinx.atomicfu.*
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.annotations.Param
+import org.jetbrains.kotlinx.lincheck.checkImpl
+import org.jetbrains.kotlinx.lincheck.paramgen.IntGen
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.junit.Test
+
+/**
+ * This test is created to verify this [bug](https://github.com/JetBrains/lincheck/issues/209) is resolved.
+ * The problem was that we ran in infinite spin-lock in trace collection mode due to incorrect Loop detector work.
+ */
+@Param(name = "element", gen = IntGen::class, conf = "0:3")
+class BoundedQueueTest {
+    private val queue = QueueAdaptor<Int>()
+
+    @Operation
+    fun enqueue(@Param(name = "element") element: Int) = queue.enqueue(element)
+
+    @Operation
+    fun dequeue() = queue.dequeue() != null
+
+    @Test
+    fun modelCheckingTest() =
+        ModelCheckingOptions()
+            .iterations(100)
+            .invocationsPerIteration(5_000)
+            .actorsBefore(2)
+            .threads(3)
+            .actorsPerThread(2)
+            .actorsAfter(2)
+            .checkObstructionFreedom(false)
+            .minimizeFailedScenario(false)
+            .sequentialSpecification(BoundedIntQueueSequential::class.java)
+            .checkImpl(this::class.java)
+            .checkLincheckOutput("bounded_queue_incorrect_results.txt")
+
+}
+
+@Suppress("unused")
+class BoundedIntQueueSequential {
```

Comment:
I don't think you need it

Reply:
Why do you need it?

Reply:
Finally removed, somehow didn't notice it twice.